src/orion/core/utils/__init__.py

```diff
@@ -48,9 +60,14 @@ def __init__(cls, name, bases, dictionary):
     def __call__(cls, *args, **kwargs):
         """Create an object if does not already exist, otherwise return what there is."""
         if cls.instance is None:
-            cls.instance = super(SingletonType, cls).__call__(*args, **kwargs)
+            try:
+                cls.instance = super(SingletonType, cls).__call__(*args, **kwargs)
+            except TypeError:
+                raise SingletonNotInstantiatedError(cls.__name__)
```

Comment:
This will mute the stack above and make it difficult to debug. Should `raise <custom error> from e`.